src/wrappers/PhylorefWrapper.js

```diff
@@ -643,49 +660,91 @@ class PhylorefWrapper {
     if (!has(phylorefAsJSONLD, '@id')) phylorefAsJSONLD['@id'] = fallbackIRI;
     phylorefAsJSONLD['@type'] = 'owl:Class';
 
-    // All phyloreferences are subclasses of phyloref:Phyloreference.
-    phylorefAsJSONLD.subClassOf = 'phyloref:Phyloreference';
-
-    // Construct an equivalentClass expression for this phyloreference.
+    // Construct a class expression for this phyloreference.
     const internalSpecifiers = phylorefAsJSONLD.internalSpecifiers || [];
     const externalSpecifiers = phylorefAsJSONLD.externalSpecifiers || [];
 
+    // In the following section: we ran into a bug (https://github.com/phyloref/phyx.js/issues/57)
+    // that was caused by OWL reasoners deciding that two phylorefs were identical
+    // because they shared an equivalentClass expression. To prevent this from
+    // happening, we now set up each phyloreference as a subclass of every logical
+    // expression that evaluates to it. We use the createAdditionalClass() method
```

Comment:
So here "additional class" means a subclass? Again, another reason it seems to clean up and make the terminology more specific?

Reply:
Before this PR, additional classes were used only for caching subexpressions: when building a large logical expression, the same subexpressions can come up multiple times, and writing out the full logical expression caused OWL reasoners to slow down dreadfully. So I built the additional class system so that subexpressions were turned into their own OWL classes, and included in the largest expression by reference. This sped up reasoning and made the logical expressions a bit more readable.

In this PR, we're using the additional class system for another purpose: when a phyloref can be represented by multiple logical expressions, we no longer say that it is equivalent to all of them. Instead, each logical expression is turned into a subclass of the phyloref. This is pretty much a hack to get around the our inability to use unions in OWL 2 EL. The phyloref therefore includes all the instances matched by each logical expression without the OWL reasoner incorrectly inferring that these expressions are equivalent to each other.

I like your idea of renaming additional classes to component classes, and I think that terminology works for both cached and "union" component classes, so I've done that in 607405f. If you think it makes more sense to split up these two kinds of component classes, let me know!

Reply:
Sounds OK, though I'm wondering if these are generated subclasses, why not call them that.

Reply:
This is true, but note that the subclass relationships are the other way around for cached and "union" component classes.

For cached component classes, the component class is a superclass of some of the logical representations of the phyloreference, which are in the form [component class] excludes [specifier]. For example, some of the logical expressions of Crocodyloidae are in the form [(includes *C. niloticus* excludes *A. mississippiensis*) excludes *G. gangeticus*]. The component class (represented by the round brackets) is a superclass of Crocodyloidae, since every resolved node must also be a member of the class represented by (includes *C. niloticus* excludes *A. mississippiensis*). (This is taken from the [Brochu 2003](https://github.com/phyloref/phyx.js/blob/607405f7b3fb7b1adda8de42da998f2a14ea2416/test/examples/correct/brochu_2003.nq) example).

"Union" component classes are subclasses of their phyloreference. Crocodyloidae can be represented as the union of four logical expressions, and is the parent class of each of them.